godot-macros/src/class/data_models/func.rs

```diff
@@ -178,9 +192,13 @@ fn make_forwarding_closure(class_name: &Ident, signature_info: &SignatureInfo) -
     }
 }
 
-fn get_signature_info(signature: &venial::Function) -> SignatureInfo {
+fn get_signature_info(signature: &venial::Function, gd_self: bool) -> SignatureInfo {
     let method_name = signature.name.clone();
-    let mut receiver_type = ReceiverType::Static;
+    let mut receiver_type = if gd_self {
+        ReceiverType::GdSelf
+    } else {
+        ReceiverType::Static
+    };
```

Comment:
With this logic, the receiver type could technically be overwritten when there is a `&self` or `&mut self` parameter.

So you could declare the following, and `gd_self` would be ignored:
```rs
#[func(gd_self)]
fn function(&self, arg: i32) { ... }
```

Could you validate against this with `bail!` and a meaningful error message?

Reply:
If I check that here a bunch of stuff needs to be changed because trying to propagate an error up from here causes a massive headache.
There's a bunch of places that transitively call this that aren't set up to handle errors. Like when this is called for generated methods.
Should I not check this here, and rely on the check that happens higher up the pipeline when removing the first element from the signature?

Reply:
> If I check that here a bunch of stuff needs to be changed because trying to propagate an error up from here causes a massive headache. There's a bunch of places that transitively call this that aren't set up to handle errors. Like when this is called for generated methods. Should I not check this here, and rely on the check that happens higher up the pipeline when removing the first element from the signature?

That's also possible, but then I would:
* make sure that check actually happens on the higher level 🙂 
* write a short comment here that this condition is verified higher up (and where)

Reply:
I actually changed this so that now the `Gd<Self>` argument isn't deleted from the signature, it's replace with `&self`. That way gd_self is checked alongside the other reference types so there's no potential logic error.